pkg/acsengine/engine.go

```diff
@@ -811,6 +818,27 @@ func addSecret(m paramsMap, k string, v interface{}, encode bool) {
 	addKeyvaultReference(m, k, parts[1], parts[2], parts[4])
 }
 
+func addArraySecret(m paramsMap, k string, v interface{}, encode bool) {
+	arr, ok := v.([]string)
+	if !ok {
+		addValue(m, k, v)
+		return
+	}
+	values := make([]string, len(arr))
+	for i := 0; i < len(arr); i++ {
+		str := arr[i]
+		parts := keyvaultSecretPathRe.FindStringSubmatch(str)
+		if parts == nil || len(parts) != 5 {
+			if encode {
+				values[i] = base64.StdEncoding.EncodeToString([]byte(str))
+			} else {
+				values[i] = str
+			}
+		}
+	}
+	addValue(m, k, values)
```

Comment:
Do you need to call addKeyvaultReference function as in the above addSecret function?

Reply:
I will look into this, not sure